### DISCUSSION TITLE
Python serverless functions exceed the limit

### DISCUSSION THREAD

**Discussion by PumPum7** [vercel/next.js#47062] [Help]
### Summary

I have recently converted my Website from React with React Router to NextJS. Serverless functions work fine locally but when I deploy them to Vercel I keep getting the error that the serverless functions exceed the limit of 50mb (sometimes they are 59mb and sometimes over 200mb depending on my vercel.json).
According to pypis size field the total size of the dependencies should be 12.41mb, the python file itself is 39.9KB. So I am not sure how it gets to 59mb
Here is the error:
![image](https://user-images.githubusercontent.com/28806828/224664101-846a7cfd-8728-4f06-8aea-35b289495f74.png)
 

### Additional information

```js
The requirements.txt file:
click==8.1.3
pathspec==0.11.0
async-timeout==4.0.2
Authlib==1.2.0
base58==2.1.1
cachelib==0.9.0
cachetools==4.2.4
click==8.1.3
construct==2.10.68
construct-typing==0.5.5
dnspython==2.3.0
Flask==2.2.3
Flask-Caching==2.0.2
Flask-Session==0.4.0
h11==0.14.0
httpcore==0.16.3
httpx==0.23.3
itsdangerous==2.1.2
jsonalias==0.1.1
peewee==3.15.4
pymongo==4.3.3
PyNaCl==1.5.0
redis==4.5.1
rfc3986==1.5.0
solana==0.29.1
solders==0.14.3
types-cachetools==4.2.10
websockets==10.4
Werkzeug==2.2.3
```

**Comment by icyJoseph** [comment]
This is coming from upstream, AWS. The AWS serverless runtime has the 50MB limit. 

- https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html#function-configuration-deployment-and-execution

I wonder if the Python runtime is at play here? Do you know which one you have? Is it PyPy? Just guessing here btw, but IIRC, everything, including the runtime is taken into account in AWS' calculation.

**Reply by PumPum7** [reply]
True that could also get taken into account. I havent specified a python version or anything like that im my Pipfile so the default version (3.9) should be used. Is there any way to find out which files get build/are taken into account?
I have also not specified a source so it should be the default aswell:
```py
[[source]]
url = "https://pypi.org/simple"
verify_ssl = true
name = "pypi"
 ```

**Reply by icyJoseph** [reply]
I am not really sure. I think it might be worth opening a ticket with the support team in Vercel.com ~
